test/unit/utils/test-xhr-utils.js

```diff
@@ -0,0 +1,204 @@
+      expect(init['headers']['Accept']).to.equal('text/html');
+    });
+
+    it('should handle null credentials', () => {
+      allowConsoleError(() => { expect(() => {
+        setupInit({credentials: null}, 'text/html');
+      }).to.throw(/Only credentials=include\|omit support: null/); });
+    });
+  });
+
+  describe('setupJsonFetchInit', () => {
+    it('set proper properties', () => {
+      expect(setupJsonFetchInit({body: {}})).to.deep.equal({
+        headers: {
+          'Accept': 'application/json',
+        },
+        body: {},
+        method: 'GET',
+      });
+
+      expect(setupJsonFetchInit({body: {}, method: 'POST'})).to.deep.equal({
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'text/plain;charset=utf-8',
+        },
+        body: '{}',
+        method: 'POST',
+      });
+    });
+  });
+
+  describe('getViewerInterceptResponse', () => {
+    let viewer;
+    let viewerForDoc;
+    beforeEach(() => {
+      viewer = {
+        hasCapability: unusedParam => false,
+        whenFirstVisible: () => Promise.resolve(),
+        sendMessageAwaitResponse: sandbox.stub(),
+      };
+      viewerForDoc = sandbox.stub(Services, 'viewerForDoc');
+    });
+
+    it('should be no-op if amp doc is absent', () => {
+      return getViewerInterceptResponse({}, null, '', {}).then(() => {
+        expect(viewerForDoc).to.not.have.been.called;
+      });
+    });
+
+    it('should be no-op if amp doc does not support xhr interception', () => {
+      viewerForDoc.returns(viewer);
+      const doc = document.createElement('html');
+      const ampDoc = {
+        getRootNode: () => {
+          return {documentElement: doc};
+        },
+      };
+      return getViewerInterceptResponse({}, ampDoc, 'https://www.googz.org')
+          .then(() => {
+            expect(viewer.sendMessageAwaitResponse).to.not.have.been.called;
+          });
+    });
+
+    it('should send xhr request to viewer', () => {
+      const win = {AMP_MODE: {development: false}};
+      viewer = Object.assign(viewer, {
+        hasCapability: unusedParam => true,
+        isTrustedViewer: () => Promise.resolve(true),
+      });
+      viewerForDoc.returns(viewer);
+      const doc = document.createElement('html');
+      doc.setAttribute('allow-xhr-interception', 'true');
+      const ampDoc = {
+        getRootNode: () => {
+          return {documentElement: doc};
+        },
+      };
+      viewer.sendMessageAwaitResponse.returns(Promise.resolve({}));
+      return getViewerInterceptResponse(
+          win, ampDoc, 'https://www.googz.org', {body: {}})
+          .then(() => {
+            const msgPayload = dict({
+              'originalRequest': {
+                'input': 'https://www.googz.org',
+                'init': {
+                  'body': {},
+                },
+              },
+            });
+            expect(viewer.sendMessageAwaitResponse).to.have.been.called;
```

Comment:
`.calledOnce`. Otherwise, the `.calledWith` line below makes this line redundant.

Reply:
done.